account/src/main/java/com/caldevsupplychain/account/controller/AccountControllerImpl.java

```diff
@@ -0,0 +1,219 @@
+package com.caldevsupplychain.account.controller;
+
+import com.caldevsupplychain.account.model.User;
+import com.caldevsupplychain.account.service.AccountService;
+import com.caldevsupplychain.account.util.AccountWSBuilder;
+import com.caldevsupplychain.account.validator.EditUserValidator;
+import com.caldevsupplychain.account.validator.LoginValidator;
+import com.caldevsupplychain.account.validator.SignupValidator;
+import com.caldevsupplychain.common.bean.account.UserBean;
+import com.caldevsupplychain.common.exception.FormErrorsExceptionHandler;
+import com.caldevsupplychain.common.type.ErrorCode;
+import com.caldevsupplychain.common.ws.account.ApiErrorsWS;
+import com.caldevsupplychain.common.ws.account.ErrorWS;
+import com.caldevsupplychain.common.ws.account.UserWS;
+import com.caldevsupplychain.notification.mail.service.EmailService;
+import lombok.AllArgsConstructor;
+import lombok.extern.slf4j.Slf4j;
+import org.apache.shiro.SecurityUtils;
+import org.apache.shiro.authc.AuthenticationException;
+import org.apache.shiro.authc.UsernamePasswordToken;
+import org.apache.shiro.authz.annotation.RequiresPermissions;
+import org.apache.shiro.subject.Subject;
+import org.springframework.http.HttpStatus;
+import org.springframework.http.ResponseEntity;
+import org.springframework.validation.BindingResult;
+import org.springframework.validation.annotation.Validated;
+import org.springframework.web.bind.annotation.*;
+
+import javax.mail.MessagingException;
+import javax.servlet.http.HttpServletRequest;
+import javax.websocket.server.PathParam;
+import java.util.List;
+import java.util.Optional;
+
+
+
+@Slf4j
+@RestController
+@RequestMapping("/api/account/v1")
+@AllArgsConstructor
+public class AccountControllerImpl implements AccountController {
+
+    private AccountService accountService;
+
+    private AccountWSBuilder accountWSBuilder;
+
+    /* validators */
+    private SignupValidator signupValidator;
+    private LoginValidator loginValidator;
+    private EditUserValidator editUserValidator;
+
+    private EmailService emailService;
+
+    /* exception handler for form fields and bind field errors together */
+    private FormErrorsExceptionHandler formErrorsExceptionHandler;
+
+
+    /************************************************************************************************
+     |									Account API													|
+     ************************************************************************************************/
+
+    @PostMapping("/signup")
+    public ResponseEntity<?> signup (
+            @RequestParam(required = false, defaultValue = "USER") String role,
+            @Validated @RequestBody UserWS userWS,
+            HttpServletRequest request,
```

Comment:
Don't think `request` is needed here. Can you remove it?

Reply:
fixed. Removed it and the email system now find the email template by the type of the email and have preset activation link, or subject title etc.